packages/hop-node/src/watchers/PolygonBridgeWatcher.ts

```diff
@@ -55,78 +59,47 @@ class PolygonBridgeWatcher extends BaseWatcher {
       this.chainId === this.polygonMainnetChainId ? 'matic' : 'mumbai'
     }/block-included`
 
-    this.maticPOSClient = new MaticPOSClient({
-      network: this.chainId === this.polygonMainnetChainId ? 'mainnet' : 'testnet',
-      version: this.chainId === this.polygonMainnetChainId ? 'v1' : 'mumbai',
-      maticProvider: new Web3.providers.HttpProvider(
-        this.l2Provider.connection.url
-      ),
-      parentProvider: new Web3.providers.HttpProvider(
-        this.l1Provider.connection.url
-      )
-    })
+    this.maticClient = new FxPortalClient()
+
+    this.init()
+      .catch((err: any) => {
+        this.logger.error('matic client initialize error:', err)
+      })
   }
 
-  async start () {
-    this.logger.debug(`polygon ${this.tokenSymbol} bridge watcher started`)
-    this.started = true
-    try {
-      const l2TokenAddress =
-        globalConfig.addresses[this.tokenSymbol][Chain.Polygon]?.l2CanonicalToken
-      if (!l2TokenAddress) {
-        throw new Error(
-          `no token address found for ${this.tokenSymbol} on ${Chain.Polygon}`
-        )
-      }
-      const l2Token = new Contract(l2TokenAddress, erc20Abi, this.l2Wallet)
-
-      const transactionHashes: any = {}
-      l2Token
-        .on(
-          'Transfer',
-          (sender: string, to: string, data: string, event: Event) => {
-            const { transactionHash } = event
-            if (to === constants.AddressZero) {
-              this.logger.debug(
-                'received transfer event. tx hash:',
-                transactionHash
-              )
-              transactionHashes[transactionHash] = event
-            }
-          }
-        )
-        .on('error', this.logger.error)
-
-      while (true) {
-        if (!this.started) {
-          return
+  async init () {
+    const bonder = await this.l1Wallet.getAddress()
+    const rootTunnel = globalConfig.addresses[this.tokenSymbol][Chain.Polygon].l1FxBaseRootTunnel
+    await this.maticClient.init({
+      network: this.chainId === this.polygonMainnetChainId ? 'mainnet' : 'testnet',
+      version: this.chainId === this.polygonMainnetChainId ? 'v1' : 'mumbai',
+      parent: {
+        provider: this.l1Wallet,
+        defaultConfig: {
+          from: bonder
```

Comment:
Do you understand why this is here? It is not the case that the `commitTransfer` functions will all come from the bonder address. Do you know if this init expects accurate `from` addresses for each `commitTransfers` and exit transactions?

As a data point, in the test exits we performed, the commits were performed by [`0xfefe...`](https://polygonscan.com/tx/0x70bb6dbee84bd4ef1cd1891c666733d0803d81ac762ff7fdc4726e4525c1e23b) (not a bonder) and [`0xad1...`](https://polygonscan.com/tx/0xb2d6e5886402782aa0adfe2d96b5c874fe7501b892b88cae3e4b9dddba962f86) (a bonder). The exits were performed by [`0x2a6...`](https://etherscan.io/tx/0x3d2c8703fe04d5ff05bad7829415bc34d308e12f568f949334060adca9ff344e) (not a bonder on this bridge set).

Reply:
Without the from address here, the matic library throws an error (see screenshot). I renamed the variable to simply `from`

![DeepinScreenshot_select-area_20220125125611](https://user-images.githubusercontent.com/168240/151058600-f7e702fa-0d1f-4324-9976-be27a2167cac.png)

Resolved with https://github.com/hop-protocol/hop/pull/384/commits/fef0c0adff680b369340f9b219ca3b5acbd002b8